src/main/java/org/example/daos/AuthDao.java

```diff
@@ -0,0 +1,80 @@
+package org.example.daos;
+
+import org.example.exceptions.DatabaseConnectionException;
+import org.example.exceptions.DoesNotExistException;
+import org.example.exceptions.Entity;
+import org.example.models.LoginRequest;
+import org.example.models.User;
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class AuthDao {
+
+    public User getUser(final LoginRequest loginRequest, final Connection c)
+            throws SQLException, DatabaseConnectionException,
+            DoesNotExistException {
+            String query = "SELECT * FROM `User` "
+                  +  "WHERE Email = ?";
+            PreparedStatement statement = c.prepareStatement(query);
+
+            statement.setString(1, loginRequest.getEmail());
+
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                if (BCrypt.checkpw(loginRequest.getPassword(),
+                        resultSet.getString("Hash"))) {
+                    return new User(
+                            resultSet.getString("Email"),
+                            resultSet.getString("Salt"),
+                            resultSet.getString("Hash"),
+                            resultSet.getInt("RoleId"));
+                } else {
+                    return null;  //user found but wrong password
+                }
+            }
+        throw new DoesNotExistException(Entity.USER); //user not found
+    }
+
+    public void generateUsers(final Connection c) throws SQLException,
```

Comment:
Is this needed in the live version, This is test code isn't it?

Reply:
this is needed